### DISCUSSION TITLE
Code snippet plugin has incorrect diff

### DISCUSSION THREAD

**Discussion by Nixinova** [lowlighter/metrics#875] [Q&A] [answered]
The code snippet plugin shows this:
![image](https://user-images.githubusercontent.com/42429413/153735156-00ff9b11-f51d-4c81-ba71-8bd4ea729d94.png)
but the actual diff is this: 
![image](https://user-images.githubusercontent.com/42429413/153735158-6d56d1b5-7668-4054-b5de-d00d4fd3fb78.png) https://github.com/Nixinova/Inlay-Regex/commit/4b825a8f654f739801dd9e499e0ddd3594ca6e8a

the last line is erroneous in the plugin

**Comment by spenserblack** [comment, marked as answer]
I'm guessing what happened is that, because the last line had a `-`
```markdown
- foo
```

it was interpreted as the `-` in a diff.
```diff
- foo
```
🤣 

I think the solution is that the code snippet plugin should precede any unchanged line with a space (` `), which is how git shows diffs.
```diff
+- addition
 - foo
--deletion
```

**Reply by lowlighter** [reply]
Indeed yes 😆 !
